src/Agent/NewRelic/Agent/Core/DistributedTracing/TracingState.cs

```diff
@@ -148,45 +149,30 @@ public string TransactionId
             }
         }
 
+        private bool? _sampled;
         public bool? Sampled
         {
             get
             {
-                if (_sampled != null && _sampled.HasValue)
-                {
+                if (_sampled is not null)
                     return _sampled;
-                }
 
-                if (_traceContext?.Tracestate != null && _traceContext.Tracestate.Sampled.HasValue)
-                {
+                if (_traceContext?.Tracestate is { Sampled: not null })
                     return _sampled = Convert.ToBoolean(_traceContext.Tracestate.Sampled);
-                }
 
-                if (_newRelicPayload?.Sampled != null && _newRelicPayload.Sampled.HasValue)
-                {
+                if (_newRelicPayload?.Sampled is not null)
                     return _sampled = _newRelicPayload.Sampled;
-                }
 
                 return null;
             }
         }
 
+        private float? _priority;
         public float? Priority
         {
-            get
-            {
-                if (_traceContext?.Tracestate != null)
-                {
-                    return _traceContext.Tracestate.Priority;
-                }
-
-                if (_newRelicPayload?.Priority != null && _newRelicPayload.Priority.HasValue)
-                {
-                    return _newRelicPayload.Priority;
-                }
-
-                return null;
-            }
+            get => _priority ??
```

Comment:
These nested conditionals seem harder to read, and might make it harder to step through with a debugger. I think that it might be better to split up this logic using multiple if blocks and return statements (similar to the previous version of this code).

Reply:
I like it the way it is, but I'll change it since the implementation is a bit inconsistent with other properties in this class..